### PR TITLE
S122 direct lambda call fixes

### PR DESCRIPTION
**Pull request by jlorper**
- Grant permissions to proxy caller to invoke lambdas
- Removes gateway configuration
- Removes concat call in resources names
- Test function for AWS sign calls with "lambda" service
- NPE in handler when switching from API Gateway
- Last commit is also tag "v0.3.0-beta.1", pointed everything to this, so passes CI

### Change implications

 - dependencies added/changed? **no**

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202219382641155

### CODE REVIEW

**Review by aperez-worklytics — APPROVED** [review]

File: infra/examples/aws-google-workspace/main.tf [hunk 1]
```diff
@@ -49,7 +49,7 @@ module "psoxy-package" {
 
 # holds SAs + keys needed to connect to Google Workspace APIs
 resource "google_project" "psoxy-google-connectors" {
-  name            = "Worklytics Connectors${var.environment_name == null ? "" : concat(" - ", var.environment_name)}"
```
Comment (eschultink) [inline infra/examples/aws-google-workspace/main.tf:52]: thanks, my fault.

File: tools/test-psoxy.sh [hunk 2]
```diff
@@ -45,7 +45,7 @@ then
   log "Calling proxy..."
   log "Request: $TEST_URL"
   log "Waiting Response:"
-  awscurl "${curlparams[@]}" --service execute-api --access_key $CALLER_ACCESS_KEY_ID --secret_key $CALLER_SECRET_ACCESS_KEY --security_token $CALLER_SESSION_TOKEN $TEST_URL
```
Comment (eschultink) [inline tools/test-psoxy.sh:48]: 👍 ; ok, yeah, I see that the auth signature is coupled to the service, not just the URL.  That's very strange to me (why should caller be aware of the AWS service used the implement a REST API??), but it's AWS fault not ours.

Comment (eschultink) [reply tools/test-psoxy.sh:48]: direct calls to the lambdas is the KISS solution that I think will work for 90% of customers, but customer call from Friday does get me worried that API gateways will be appealing to some enterprise customers precisely BC of the complexity and the ability to put a bunch of additional rules/restrictions on the mappings.

File: infra/examples-dev/aws-msft-365/main.tf [hunk 3]
```diff
@@ -177,11 +177,11 @@ module "psoxy-msft-connector" {
 
   function_name        = "psoxy-${each.key}"
   source_kind          = each.value.source_kind
-  api_gateway          = module.psoxy-aws.api_gateway
   path_to_function_zip = module.psoxy-package.path_to_deployment_jar
   function_zip_hash    = module.psoxy-package.deployment_package_hash
   path_to_config       = "../../../configs/${each.value.source_kind}.yaml"
   api_caller_role_arn  = module.psoxy-aws.api_caller_role_arn
+  api_caller_role_arn_name = module.psoxy-aws.api_caller_role_name
```
Comment (eschultink) [inline infra/examples-dev/aws-msft-365/main.tf:184]: style (`terraform fmt -recursive`)

File: infra/examples/aws-google-workspace/main.tf [hunk 4]
```diff
@@ -33,23 +33,23 @@ provider "aws" {
 }
 
 module "psoxy-aws" {
-  source = "git::https://github.com/worklytics/psoxy//infra/modules/aws?ref=v0.2.0-beta.1"
+  source = "git::https://github.com/worklytics/psoxy//infra/modules/aws?ref=v0.3.0-beta.1"
```
Comment (eschultink) [inline infra/examples/aws-google-workspace/main.tf:36]: +1

**Review by eschultink — APPROVED** [review]
👍  looks good. one idea is to add a `CHANGELOG` at top level to clarify that `0.2` is initial tagged version, and `0.3` does the direct lambda change.

**Pull request merged**